app/components/select/item.tsx

```diff
@@ -0,0 +1,64 @@
+import React from "react"
+import { ActivityIndicator, View } from "react-native"
+
+import { ListItem, ListItemProps, makeStyles, useTheme } from "@rneui/themed"
+import Icon from "react-native-vector-icons/Ionicons"
+
+const useStyles = makeStyles(() => ({
+  iconContainer: { width: 18 },
+}))
+
+export type ItemProps = {
+  children: React.ReactNode
+  value: string
+  selected: boolean
+  onChange: (optionKey: string) => void | Promise<void>
+  loading: boolean
+  setLoading: React.Dispatch<React.SetStateAction<boolean>>
+} & ListItemProps
+
+export const Item: React.FC<ItemProps> = ({
+  children,
+  value,
+  selected,
+  onChange,
+  loading,
+  setLoading,
+  ...props
+}) => {
+  const {
+    theme: { colors },
+  } = useTheme()
+  const styles = useStyles()
+
+  const [showActivityIndicator, setShowActivityIndicator] = React.useState(false)
+
+  const onPress = () => {
+    if (selected || loading) return
+
+    const isPromise = onChange(value)
```

Comment:
this variable name is misleading. a `isPromise` would appear at first to be a boolean - because of the `is`. It seems that's not the case here.

Reply:
I've removed it fully